src/Mollie/WC/Helper/IconFactory.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+use Mollie\Api\Types\PaymentMethod;
+
+class Mollie_WC_Helper_IconFactory
+{
+
+    /**
+     * @param $data
+     * @return string
+     */
+    public function create($data)
+    {
+        if (!preg_match('/[^A-Za-z0-9]/', $data)) {
+            $methods = $this->getApiMethods();
+            if(!$methods){
+                return $this->fallbackToAssetsFolder($data);
+            }
+
+            foreach ($methods as $method){
+                if($method['id'] == $data){
+                    return $method['image']->svg;
+                }
+            }
```

Comment:
I see performances problems in this approach. You have to loop every time the same list to find the correct item.
In a checkout page that means that if you have all of the gateways enabled you have to loop through the `$methods` N times, where `N` is the number of gateway displayed in the page. 

If you inject the methods at construction time you can inject a different methods list where the `id` is the key of the item, allowing you to access directly the item for which you want to retrieve the svg.

Eg. 

```
$methods = [
    'methodID' => [
         'svg' => '',
         // ...
    ] 
];
```

I would suggest to use `array_column` to retrieve the `id` of the methods and then `array_combine` to build the list.

See https://3v4l.org/UjDg2